internal/pkg/deploy/cloudformation/stack/validate.go

```diff
@@ -100,6 +114,154 @@ func validateSidecarMountPoints(in []manifest.SidecarMountPoint) error {
 	return nil
 }
 
+func validateSidecarDependsOn(in manifest.SidecarConfig, sidecarName string, s convertSidecarOpts) error {
+	if in.DependsOn == nil {
+		return nil
+	}
+
+	for name, status := range in.DependsOn {
+		// status must be one of < start | complete | success >
+		if !isValidStatus(status) {
+			return errInvalidDependsOnStatus
+		}
+		// essential containers must have 'start' as a status
+		if name == s.workloadName || s.sidecarConfig[name].Essential == nil || aws.BoolValue(s.sidecarConfig[name].Essential) {
```

Comment:
Can we wrap this complex if statement in a small function  so that it's easier to read? that way we can write:
```go

func (s *convertSidecarOpts) isEssential(containerName string) bool {...}

if s.isEssential(name) && status != dependsOnStart {
    return errEssentialContainerStatus
}
```

This also avoids the more deeply nested conditional, which can be difficult to read and reason about.